identity/README.md

```diff
@@ -0,0 +1,299 @@
+| CSI        | [Container Storage Interface][csi_spec].                                                 |
+| CSI-driver | A component that implements CSI service RPCs, the CSI Controller or NodePlugin.          |
+| CO         | Container Orchestration system that communicates with users and CSI-Addons service RPCs. |
+| SP         | Storage Provider, the vendor of a CSI-driver implementation.                             |
+| RPC        | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).            |
+
+## Objective
+
+Define an API that makes it possible for the CSI-Addons Controller(s) to
+communicate with the CSI-Addons side-cars that are running with the CSI-driver
+components.
+
+### Goals in MVP
+
+The API will
+
+* make it possible for worker nodes (CSI-Addons side-car) to announce
+  themselves
+* have the CSI-Addons Controller list all announced CSI-Addons side-cars and
+  CSI-drivers
+* provide details about CSI-Addons features the CSI-driver supports
+
+### Non-Goals in MVP
+
+This will be limited to node discovery functionality only. CSI-Addons features
+will be documented in their own specifications. The available capabilities will
+get extended when new CSI-Addons operations are proposed.
+
+## Solution Overview
+
+CSI-Addons side-cars for the CSI-Controller and CSI-NodePlugins will announce
+themselves at the CO.
+
+The CSI-Addons Controller reacts on events generated by the CO (e.g. Kubernetes
+CRDs) that trigger actions. When a CSI-Addons Node is announced to the CO, the
+CSI-Addons Controller will connect to the Node and request the available
+capabilities.
+
+The CSI-Addons side-cars communicate over a UNIX-Domain-Socket with the
+CSI-drivers (CSI Controller and NodePlugin).
+
+## Architecture
+
+A user (someone that creates/consumes CSI volumes) instructs the CSI-Addons
+Controller to invoke a storage maintenance operation. This interaction is done
+in a CO specific way (e.g. a Custom Resourse will be provided for Kubernetes).
+
+The CSI-Addons Controller inspects the instruction from the user, and formats
+it into an operation that is sent to one or more CSI-Addons side-cars. The
+side-cars eventually invoke the operation at the CSI-driver.
+
+```text
+.------.       .------------.
+| User |-------| CSI-Addons |
+'------'       | Controller |
+               '------------'
+                      |
+                      | gRPC
+                      |
+            .---------+------------------------------.
+            |         |                              |
+            |  .------------.        .------------.  |
+            |  | CSI-Addons |  gRPC  |    CSI     |  |
+            |  |  side-car  |--------| Controller |  |
+            |  '------------'        | NodePlugin |  |
+            |                        '------------'  |
+            | CSI-driver Pod                         |
+            '----------------------------------------'
```

Comment:
this looks to be more kubernetes specific however I feel the addons should be more about the specification of the protocol between a CO and a Plugin. its up to the CO to implement/design as per the specifications